shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/resources/ManagedProcessorStatus.java

```diff
@@ -0,0 +1,108 @@
+package com.redhat.service.smartevents.shard.operator.v2.resources;
+
+import java.util.HashSet;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.redhat.service.smartevents.infra.core.models.ManagedResourceStatus;
+import com.redhat.service.smartevents.shard.operator.core.resources.Condition;
+import com.redhat.service.smartevents.shard.operator.core.resources.ConditionReasonConstants;
+import com.redhat.service.smartevents.shard.operator.core.resources.ConditionStatus;
+import com.redhat.service.smartevents.shard.operator.core.resources.ConditionTypeConstants;
+import com.redhat.service.smartevents.shard.operator.core.resources.CustomResourceStatus;
+import com.redhat.service.smartevents.shard.operator.core.utils.DeploymentStatusUtils;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.internal.readiness.Readiness;
+
+public class ManagedProcessorStatus extends CustomResourceStatus {
+
+    public static final String SECRET_AVAILABLE = "SecretAvailable";
+    public static final String IMAGE_NAME_CORRECT = "ImageNameCorrect";
+    public static final String DEPLOYMENT_AVAILABLE = "DeploymentAvailable";
+    public static final String SERVICE_AVAILABLE = "ServiceAvailable";
+    public static final String SERVICE_MONITOR_AVAILABLE = "ServiceMonitorAvailable";
+
+    private static final HashSet<Condition> EXECUTOR_CONDITIONS = new HashSet<>() {
+        {
+            add(new Condition(ConditionTypeConstants.READY, ConditionStatus.Unknown));
+            add(new Condition(SECRET_AVAILABLE, ConditionStatus.Unknown));
+            add(new Condition(IMAGE_NAME_CORRECT, ConditionStatus.Unknown));
+            add(new Condition(DEPLOYMENT_AVAILABLE, ConditionStatus.Unknown));
+            add(new Condition(SERVICE_AVAILABLE, ConditionStatus.Unknown));
+            add(new Condition(SERVICE_MONITOR_AVAILABLE, ConditionStatus.Unknown));
+        }
+    };
+
+    public ManagedProcessorStatus() {
+        super(EXECUTOR_CONDITIONS);
+    }
+
+    @Override
+    public ManagedResourceStatus inferManagedResourceStatus() {
```

Comment:
Not needed as the shard now only reports the `status` of the resource back to the Control Plane